e2e/test_type_without_auto_space.py

```diff
@@ -0,0 +1,100 @@
+
+
+class TestAppium(unittest.TestCase):
+    def setUp(self):
+        self.driver = webdriver.Remote(appium_server_url, options=options)
+        self.driver.implicitly_wait(15)
+
+    def tearDown(self):
+        self.disable_hush_keyboard()
+
+        if self.driver:
+            self.driver.quit()
+
+    def test_switch_default_keyboard_to_hush_keyboard(self):
+        self.enable_hush_keyboard()
+
+        self.switch_hush_keyboard_to_input_keyboard()
+
+        self.turn_off_the_auto_space()
+
+        self.enter_R_U_F_L_D_B_in_sequence()
+
+    def disable_hush_keyboard(self):
+        # switch Hush Keyboard to disable
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@text="Enable keyboard"]')
+        el.click()
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Hush Keyboard"]')
+        el.click()
+
+    def enable_hush_keyboard(self):
+        # click Enable keyboard bottom
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@text="Enable keyboard"]')
+        el.click()
+        # check Hush Keyboard is disabled and switch to enable
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Hush Keyboard"]')
+        self.assertEqual(el.get_attribute('checked'), 'false')
+        el.click()
+        # click ok to agree notification
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@text="OK"]')
+        el.click()
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@text="OK"]')
+        el.click()
+        # check Hush Keyboard is enabled
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Hush Keyboard"]')
+        self.assertEqual(el.get_attribute('checked'), 'true')
+        # return to previous page
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Navigate up"]')
+        el.click()
+
+    def switch_hush_keyboard_to_input_keyboard(self):
+        # click Select input method bottom and select Hush Keyboard
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@text="Select input method"]')
+        el.click()
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@resource-id="android:id/text1" and @text="Hush '
+                                                               'Keyboard"]/parent::*')
+        el.click()
+
+    def turn_off_the_auto_space(self):
+        # enter to settings page
+        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Settings"]')
+        el.click()
+
+
+        # check Add space after notation is off
+        el = self.driver.find_element(by=AppiumBy.XPATH,
+                                      value='//*[@text="Add space after notation"]/preceding-sibling::*['
+                                            '@checkable="true"]')
+        el.click()
+        self.assertEqual(el.get_attribute('checked'), 'false')
+        self.driver.back()
```

Comment:
Please help me remove double blank lines.

```suggestion
    def turn_off_the_auto_space(self):
        # enter to settings page
        el = self.driver.find_element(by=AppiumBy.XPATH, value='//*[@content-desc="Settings"]')
        el.click()

        # check Add space after notation is off
        el = self.driver.find_element(by=AppiumBy.XPATH,
                                      value='//*[@text="Add space after notation"]/preceding-sibling::*['
                                            '@checkable="true"]')
        el.click()
        self.assertEqual(el.get_attribute('checked'), 'false')
        self.driver.back()
```